crates/nu-protocol/src/value/mod.rs

```diff
@@ -789,7 +789,34 @@ impl Value {
 
     /// Convert Value into a debug string
     pub fn debug_value(&self) -> String {
-        format!("{self:#?}")
+        // Let's do some special handling for emojis
+        const ZERO_WIDTH_JOINER: &str = "\u{200d}";
+        const VARIATION_SELECTOR_16: &str = "\u{fe0f}";
+        const SKIN_TONES: [&str; 5] = [
+            "\u{1f3fb}", // Light Skin Tone
+            "\u{1f3fc}", // Medium-Light Skin Tone
+            "\u{1f3fd}", // Medium Skin Tone
+            "\u{1f3fe}", // Medium-Dark Skin Tone
+            "\u{1f3ff}", // Dark Skin Tone
+        ];
+
+        match self {
+            Value::String { val, .. } => {
+                if val.contains(ZERO_WIDTH_JOINER)
+                    || val.contains(VARIATION_SELECTOR_16)
+                    || SKIN_TONES.iter().any(|skin_tone| val.contains(skin_tone))
+                {
+                    // This has to be an emoji, so let's display the code points that make it up.
+                    format!(
+                        "{:#?}",
+                        Value::test_string(val.escape_unicode().to_string())
```

Comment:
I don't think we need to create a new `Value::test_string`, just `val.escape_unicode().to_string()` should be fine.

Reply:
I started with that but noticed that it needed to be a `Value::String` and not just a `String`. `debug -r` uwraps our `Value`s. So, I think it needs to be a `Value::String` since `val.escape_unicode().to_string()` does not return a nushell value.

Reply:
If I read that correctly with the `derive(Debug)` on Value this would display the `Span` as well so maybe worth reusing that from `self`.

Good idea for debugging purposes @fdncred !
As `value/mod.rs` is already really long I would prefer if the `contains_emoji` logic for `&str` would live in a dedicated function (possibly in `nu-utils`).

Reply:
Let me know if the latest changes are more of what you are looking for.